### ISSUE TITLE
Source/Destination logs do not contain class names and LABELS (no info, warn or error label)

### ISSUE THREAD

**Issue by subodh1810** [airbytehq/airbyte#17861]
Currently logs emitted by source/destination do not contain class names or labels, they look like 
```
2022-10-06 21:24:25 [44msource[0m > Found table: public.table_table
2022-10-06 21:24:25 [43mdestination[0m > Preparing staging area in destination completed for schema AIRBYTE_SCHEMA stream table_table
```

Previously they used to contain classnames as well
```
2022-07-20 22:53:24 [44msource[0m > 2022-07-20 22:53:24 [32mINFO[m i.a.i.d.i.DebeziumRecordIterator(computeNext):78 - Closing cause next is returned as null
```

Not having classnames or labels makes it difficult to understand where is the log coming from and makes debugging difficult, my guess is that this issue started after PR https://github.com/airbytehq/airbyte/pull/15668

cc @grishick

**Comment by evantahler** [comment]
My interpretation of this bug is that:
* This is a problem only for Java sources/destinations?

We should test this by:
* Run the connector in isolation (SAT?) and see if it produces the expected lots (with classnames).  Is the platform truncating the classnames, or did the connector's logger stop including them?

**Comment by erohmensing** [comment]
Grooming- without having tested, we presume: 

> This is a problem only for Java sources/destinations?

yes

> Run the connector in isolation (SAT?) and see if it produces the expected lots (with classnames). Is the platform truncating the classnames, or did the connector's logger stop including them?

We'd expect it to have the same logging issue. 

Looks like it was probably introduced when the library in `base-java` used to turn logs into json format was changed - we think this might be a java sources issue?

**Comment by evantahler** [comment]
> Looks like it was probably introduced when the library in base-java used to turn logs into json format was changed - we think this might be a java sources issue?

That's probably right.  Tagging @bleonard.  @airbytehq/connector-operations probably isn't the best team to work on some of the shared DB libraries.  I'll put this back in your backlog

**Comment by bleonard** [comment]
Change likely in one place.
Probably have to republish connectors to see it. We can focus on the Postgres one and let destination team know to update at their leisure.

**Comment by prateekmukhedkar** [comment]
Logs emitted when doing a gradle build contain the LEVEL, seeing these on command line
   ` {"type":"LOG","log":{"level":"INFO","message":"Found table: models_schema_random.models_random"}}`

**Comment by prateekmukhedkar** [comment]
@subodh1810 We can configure log4j to emit class names in logs using the location info attribute (specifically via `%C`)
, but the log4j [docs](https://logging.apache.org/log4j/2.x/manual/layouts.html#LocationInformation) warn that enabling location info is costly and will result in a performance hit. Do you think we should include class name in log messages?

Does grep on 'log.message' allow you to find the source code where the log was emitted as an alternative? I understand that you would need to select a prefix of the logged message because the tail end of the message is coming from arguments that are substituted at run time.

**Comment by bleonard** [comment]
Maybe we should move to requiring a prefix in the log helpers and pass a string of the class name or whatever else makes sense?

**Comment by prateekmukhedkar** [comment]
We are getting a logger instance like [this](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java#L92), so presumably the class name should automatically appear with all the logs. Checking why it doesn't (or what part of config is overriding it).

`private static final Logger LOGGER = LoggerFactory.getLogger(PostgresSource.class);`

**Comment by prateekmukhedkar** [comment]
This looks the file that controls how logs emitting from java based sources are formatted to Airbyte json messages, and likely place where a logger name can be added. Will try adding some relevant fields here 
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java/src/main/resources/AirbyteLogMessageTemplate.json

**Comment by prateekmukhedkar** [comment]
Got the logger name to appear in logs when running unit tests. Confirming with @subodh1810 if this format looks good. However, logs emitted from Docker container for Postgres source have a different format so checking how to achieve consistency.

**Comment by prateekmukhedkar** [comment]
I modified the log pattern in log4j2.xml config to output the name of the logger that published the log event. Expected outcome was that I should see the name of the logger (or class name) when a logger instance is instantiated, for instance using `private static final Logger LOGGER = LoggerFactory.getLogger(PostgresSource.class);`

However actual outcome is that I saw logs appear like this

```
2023-02-08 21:50:42 source > i.a.w.i.VersionedAirbyteStreamFactory Found table: public.cities
2023-02-08 21:50:42 source > i.a.w.i.VersionedAirbyteStreamFactory Found table: public.users
```
Note that`VersionedAirbyteStreamFactory` appears as the logger name however the log line `Found table..` is emitted by PostgresSource class.  Investigating how is the logger name getting substituted. 

(i.a.w.i. is simply the abbreviated package name)
